api/docker-entrypoint.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+
+info()
+{
+    echo "Info: $*"
+}
+
+warn()
+{
+    echo "Warning: $*"
+}
+
+fatal()
+{
+    echo "Fatal: $*"
+}
+
+get_couchdb_url()
+{
+  echo "http://$COUCHDB_USER:$COUCHDB_PASSWORD@$COUCHDB_SERVICE_NAME:5985"
+}
+
+wait_for_couchdb()
+{
+  local n=60
+
+  while [ "$n" -gt 0 ]; do
+    curl "`get_couchdb_url`" &>/dev/null
+    [ "$?" -eq 0 ] && return 0
+    sleep 3
+    n=$[$n-1]
+  done
+
+  return 1
+}
+
+is_setup_needed()
+{
+  ! is_existing_user 'medic-api'
+}
+
+is_existing_user()
+{
+  local user="$1"
+  local url="`get_couchdb_url`"
+  local id="org.couchdb.user:$user"
+  shift 1
```

Comment:
I am not familiar (aside from reading docs just now) with `shift`. Does it impact anything outside of this function? If so, I'd like to talk about changing this since it's undesirable to have a function call come with a side effect like that.

Reply:
It clears $1 from being the passed in parameter since we locally defined it in the function as user. I was using it in a previous commit, where the random pw is generated, but I changed that to hardcore to $4 instead of shifting. I'll remove it so we are in sync with the other functions, but adding shift to those would probably be better.